### ISSUE TITLE
Mercury PoC

### ISSUE THREAD

**Issue by pplonski** [neelmshah/mercury-poc#2]
Hi @neelmshah,

I've seen that you are using Mercury. I'm the author of the framework. I've recently made a lot of changes to it. I would love to hear your feedback. Do you plan to use Mercury in some projects? What is your use case?

I'm also working on Mercury Cloud service - you will be able setup a secure website with a few clicks and just upload notebooks. Would you like to test it?

**Comment by neelmshah** [comment]
Thanks @pplonski for reaching out!  We are assessing whether it meets the project requirements and will consider using it if it does.

As for my use case, we require the integration of an interactive notebook into a web application and whiling embeding certain variables need to be passed as query parameters in the Iframe URL & it should be accessiable notebook. 

`<iframe src="http://127.0.0.1:8000/app/sample-report/embed?userId=XXX" height="700px" width="1200px"></iframe>`

Is this possiable in current version?

Regarding Mercury Cloud service, it sounds like an interesting proposition but due to security compliance, we can't host it on cloud.

**Comment by pplonski** [comment]
Sounds like a challenging project! 

URL params are not available in the current version of Mercury. I got an issue with this feature https://github.com/mljar/mercury/issues/195. If this will be the only blocker for you, then I will quickly implement this!

You should be able to easily deploy Mercury with docker-compose ([docs](https://runmercury.com/docs/docker-compose/)). If you need any help with Mercury or deployment - I'm happy to help!

**Comment by neelmshah** [comment]
Indeed, this is the only issue blocking us. I would greatly appreciate it if you could prioritize addressing the issue outlined in https://github.com/mljar/mercury/issues/195

**Comment by pplonski** [comment]
Sure, I can speed up this one. Will try to deliver it in the next week!

**Comment by neelmshah** [comment]
@pplonski  I hope you are doing well. I was wondering if you had any updates on the progress of https://github.com/mljar/mercury/issues/195. I appreciate your efforts on this.

Please let me know if there is anything I can do to help

**Comment by pplonski** [comment]
Hi @neelmshah, 

Thank you! I hope that you are doing well too! I will implement it today :) Additionally, I would like to add `Share` button, so you can easily create URL link with values.

I'm planning to add `url_key` argument to each input widget class. For example:

```python
import mercury as mr
# define text input
name = mr.Text(label="what is your name?", url_key="name")
```

The URL for app will be:
```
https://your-server/app/notebook?name=Neel
```

if you would like to embed it with predefined values:

```
https://your-server/app/notebook/embed?name=Neel
```

The exception will be file input widget, it will not have option to set its value in URL.

**Comment by pplonski** [comment]
@neelmshah sorry I will not finish this feature today ... I have `url_key` implemented in widgets. I would like to add automatic notebook re-execution with new values from URL, I will need more time for it. Should be ready tomorrow.

**Comment by pplonski** [comment]
Hi @neelmshah,

I've just release version 2.1.3 of `mercury` package. It has `url_key` for some widgets. I need to update docs.

You can add params after `?`. Example:

```
https://server/app/my-notebook/embed?key1=123&username=Neel
```

Please let me know if it works for you.

**Comment by neelmshah** [comment]
Hello @pplonski<https://github.com/pplonski>,

I wanted to express my gratitude for promptly implementing this feature. I appreciate your effort and I'm looking forward to testing it out. Rest assured that I will inform you if I encounter any difficulties.

Thank you once again for the update and for your continuous support.
________________________________

**Comment by neelmshah** [comment]
Hello @pplonski<https://github.com/pplonski>,

I'm facing an import error in the mercury > apps.storage.views​ module while executing the mercury run <notebook>​ command with version 2.1.3.

The error message I'm receiving is:

```
 File "../env/lib/python3.10/site-packages/mercury/apps/storage/urls.py", line 3, in <module>
    from apps.storage.views.dashboardfiles import (
ModuleNotFoundError: No module named 'apps.storage.views'
```
Would you mind taking a look at this and helping me troubleshoot this issue?

________________________________

**Comment by pplonski** [comment]
Hi @neelmshah, 

apologize for this, I forgot to add directory when creating a package. It is fixed already. Please install latest version `2.1.7`.

You might need to clean database locally between runs. Please run:

```
# clean database locally
mercury run clear
```

**Comment by neelmshah** [comment]
Hello @pplonski<https://github.com/pplonski>,

It's great news that the latest version is functioning correctly. However, I do have one last inquiry Is there a way to either hide the input box or make it read-only? Our objective is to use only the values of the URL key parameters in the code and prevent them from being displayed on the screen to the end user.
________________________________

**Comment by pplonski** [comment]
Hi @neelmshah,

Please take a look at arguments for `App` in the docs https://runmercury.com/docs/input-widgets/app/ You can hide the sidebar with `show_sidebar=False` argument. But user will be able to open it and edit parameters ...

I can add `disable` argument for input widgets, so you can set widgets as `disabled` (read-only) - what do you think? They will be displayed in the sidebar, but user won't be able to change its values. It will work for you?

All the best!
Piotr

**Comment by neelmshah** [comment]
Hello @pplonski<https://github.com/pplonski>,

The input widget's disabled option is works, but having a option to hide the widget with minimal effort would further enhance its usability.


________________________________

**Comment by pplonski** [comment]
Hi @neelmshah,

I will add `hidden` and `disabled` for input widgets https://github.com/mljar/mercury/issues/237

**Comment by pplonski** [comment]
I already implemented it. You can add `hidden=True` or `disabled=True` to input widgets. It is available from version `2.1.9`. I need to update docs - I hope to do this in a week. Please let me know if it works for you.

**Comment by neelmshah** [comment]
Thanks for putting in the hard work to get this feature live in such a short timeframe. That's excellent!
________________________________
